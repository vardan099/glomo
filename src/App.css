body {
  background-color: #f2f4f9;
  font-family: 'Roboto', sans-serif;
  overflow-x: hidden; }
  body #root {
    display: flex;
    flex-direction: column;
    align-items: center; }
    body #root .App {
      display: flex;
      flex-direction: column;
      align-items: center;
      background-color: #fff;
      max-width: 1100px;
      min-width: 400px; }
      @media only screen and (min-width: 600px) {
        body #root .App {
          margin: 50px 0; } }
      body #root .App .sports-poll {
        display: flex;
        flex-direction: column;
        align-items: center;
        background-image: url("./images/shape.png");
        background-position: center;
        background-repeat: no-repeat; }
        body #root .App .sports-poll_title {
          color: #b7b6bc;
          margin-bottom: 0; }
        body #root .App .sports-poll_content {
          display: flex;
          flex-direction: row;
          align-items: flex-end;
          justify-content: space-around; }
          body #root .App .sports-poll_content > div {
            flex: 1;
            text-align: center; }
        body #root .App .sports-poll_draw .info-label {
          color: #b7b6bc;
          padding-right: 15px; }
        @media only screen and (min-width: 600px) {
          body #root .App .sports-poll_draw img {
            margin: 30px 0 45px 0; } }
        body #root .App .sports-poll img.team-logo {
          width: 75%;
          max-width: 200px; }
          @media only screen and (min-width: 600px) {
            body #root .App .sports-poll img.team-logo {
              width: 100%; } }
      body #root .App .h-line {
        width: 100%;
        border: 1px solid #b7b6bc;
        margin: 15px 0; }
      body #root .App .history {
        display: flex;
        flex-direction: column;
        align-items: center;
        margin-bottom: 50px; }
        body #root .App .history_no-votes {
          color: #b7b6bc; }
        body #root .App .history_title {
          display: flex;
          flex-direction: row;
          align-items: center;
          width: calc(100% - 110px);
          justify-content: space-between; }
          body #root .App .history_title-switcher {
            cursor: pointer;
            padding-left: 10px;
            color: blue; }
          body #root .App .history_title-info {
            color: #b7b6bc;
            font-weight: 800; }
            body #root .App .history_title-info strong {
              color: black; }
        body #root .App .history_list {
          display: flex;
          flex-direction: column;
          align-items: center;
          width: 100%;
          height: 150px;
          overflow-y: auto; }
          @media only screen and (min-width: 600px) {
            body #root .App .history_list {
              width: calc(100% - 110px); } }
          body #root .App .history_list .history-item {
            display: flex;
            flex-direction: row;
            align-items: center;
            height: auto; }
            body #root .App .history_list .history-item:nth-child(odd) {
              background-color: #d5d8e1; }
            body #root .App .history_list .history-item > div {
              flex: 1;
              display: flex;
              justify-content: center;
              text-align: center; }
            body #root .App .history_list .history-item_vote-type {
              height: 30px;
              width: 100%;
              max-width: 150px;
              line-height: 30px; }
      body #root .App .bg-home {
        background-color: #5b49cf;
        color: #fff; }
      body #root .App .bg-away {
        background-color: #edbb00;
        color: #fff; }
      body #root .App .bg-draw {
        background-color: #f2f4f9;
        color: #5b49cf; }
  body .w-100 {
    width: 100%; }
  body .vote-button {
    width: 80%;
    max-width: 250px;
    height: 45px;
    border: none;
    border-radius: 25px;
    outline: none;
    color: #fff;
    cursor: pointer;
    font-size: 18px; }
    @media only screen and (min-width: 600px) {
      body .vote-button {
        width: 100%; } }
  body .custom-scroll::-webkit-scrollbar-track {
    -webkit-box-shadow: inset 0 0 6px rgba(0, 0, 0, 0.3);
    border-radius: 10px;
    background-color: #F5F5F5; }
  body .custom-scroll::-webkit-scrollbar {
    width: 6px;
    background-color: #F5F5F5; }
  body .custom-scroll::-webkit-scrollbar-thumb {
    border-radius: 10px;
    -webkit-box-shadow: inset 0 0 6px rgba(0, 0, 0, 0.3);
    background-color: #5b49cf; }

/*# sourceMappingURL=App.css.map */
