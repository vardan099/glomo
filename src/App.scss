@import "styles/colors";
@import "styles/mixin";
@import "styles/vars";

body{
  background-color:$body-background ;
  font-family: 'Roboto', sans-serif;
  overflow-x: hidden;
  #root{
    @include flex(column,center);
    .App {
      @include flex(column,center);
      background-color: $white;
      max-width: $container-max-with;
      min-width: $container-min-with;
      @media only screen and (min-width: 600px) {
        margin: 50px 0;
      }
      .sports-poll{
        @include flex(column,center);
        background-image: url('./images/shape.png');
        background-position: center;
        background-repeat: no-repeat;
        &_title{
          color: $font-second-color;
          margin-bottom: 0;
        }
        &_content{
          @include flex(row,flex-end);
          justify-content: space-around;
          &>div{
            flex: 1;
            text-align: center;
          }
        }
        &_draw{
          .info-label{
            color: $font-second-color;
            padding-right: 15px;
          }
          img{
            @media only screen and (min-width: 600px) {
              margin: 30px 0 45px 0;
            }
          }
        }
        img.team-logo{
          width: 75%;
          max-width: 200px;
          @media only screen and (min-width: 600px) {
            width: 100%;
          }
        }
      }

      .h-line{
        width: 100%;
        border: 1px solid $font-second-color;
        margin: 15px 0;
      }
      .history{
        &_no-votes{
          color: $font-second-color;
        }
        &_title{
          display: flex;
          flex-direction: row;
          align-items: center;
          width: calc(100% - 110px);
          justify-content: space-between;
          &-switcher{
            cursor: pointer;
            padding-left:10px;
            color: blue;
          }
          &-info{
            color: $font-second-color;
            font-weight: 800;
            strong{
              color: black;
            }
          }
        }
        @include flex(column,center);
        margin-bottom: 50px;
        &_list{
          @include flex(column,center);
          width: 100%;
          @media only screen and (min-width: 600px) {
            width: calc(100% - 110px);
          }
          height: 150px;
          overflow-y: auto;
          .history-item{
            @include flex(row,center);
            height: auto;
            &:nth-child(odd){
              background-color: $history-item-bg;
            }
            &>div{
              flex:1;
              display: flex;
              justify-content: center;
              text-align: center;
            }
            &_vote-type{
              height: 30px;
              width: 100%;
              max-width: 150px;
              line-height: 30px;

            }
          }
        }

      }



      .bg-home{
        background-color: $home-team-color;
        color: $white;
      }
      .bg-away{
        background-color: $away-team-color;
        color: $white;
      }
      .bg-draw{
        background-color: $draw-color;
        color: $home-team-color;
      }
    }
  }
  .w-100{
    width: 100%;
  }
  .vote-button{
    @media only screen and (min-width: 600px) {
      width: 100%;
    }
    width: 80%;
    max-width: 250px;
    height: 45px;
    border: none;
    border-radius: 25px;
    outline: none;
    color: $white;
    cursor: pointer;
    font-size: 18px;

  }

  .custom-scroll::-webkit-scrollbar-track
  {
    -webkit-box-shadow: inset 0 0 6px rgba(0,0,0,0.3);
    border-radius: 10px;
    background-color: #F5F5F5;
  }

  .custom-scroll::-webkit-scrollbar
  {
    width: 6px;
    background-color: #F5F5F5;
  }

  .custom-scroll::-webkit-scrollbar-thumb
  {
    border-radius: 10px;
    -webkit-box-shadow: inset 0 0 6px rgba(0,0,0,.3);
    background-color: $home-team-color;
  }
}



