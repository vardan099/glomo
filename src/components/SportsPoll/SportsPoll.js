import React from 'react';
import PropTypes from "prop-types";
import TeamInfo from "../TeamInfo/TeamInfo";
import Draw from "../Draw/Draw";

import homeTeamLogo from '../../images/home-logo.png'
import awayTeamLogo from '../../images/away-logo.png'


const SportsPoll = ({game, voteHandler}) => {
    const {sport, homeName, objectId, country, state, awayName, group} = game;
    return (
        <div className="sports-poll w-100">
            <h1 className="sports-poll_title">Which Team Would Win?</h1>
            <div className="sports-poll_content w-100">
                <TeamInfo
                    voteHandler={voteHandler}
                    objectId={objectId}
                    type="home"
                    teamName={homeName}
                    teamLogo={homeTeamLogo}
                />
                <Draw
                    country={country}
                    state={state}
                    voteHandler={voteHandler}
                    objectId={objectId}
                    sport={sport}
                    group={group}
                    type="draw"
                />
                <TeamInfo
                    voteHandler={voteHandler}
                    objectId={objectId}
                    type="away"
                    teamName={awayName}
                    teamLogo={awayTeamLogo}
                />
            </div>
        </div>
    )
};

SportsPoll.propTypes = {
    game: PropTypes.object,
    voteHandler: PropTypes.func
};

export default SportsPoll;