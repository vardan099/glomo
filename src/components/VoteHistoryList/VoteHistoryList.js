import React, {useState, useEffect} from 'react';
import PropTypes from "prop-types";
import VoteHistoryItem from '../VoteHistoryItem/VoteHistoryItem'

const VoteHistoryList = (props) => {
    const {polls, history} =  props;
    const [voteList,setVoteList] = useState([]);
    const [showAll,setShowAll] = useState(false);

    useEffect(() => {
        let voteList;
        if(showAll){
            voteList = [...history].reverse();
        }else{
            voteList = history.slice(-5).reverse();
        }
        setVoteList(voteList);
    }, [history, showAll]);


    const handleSeeAllClick = () =>{
        setShowAll(!showAll)
    };
    return(
        <div className="history w-100">
            <div className="history_title">
                <h3>History</h3>
                <p className="history_title-info">Count of votes: <strong>{history.length}</strong><span className='history_title-switcher' onClick={handleSeeAllClick}>{showAll?'See less':'See all'}</span></p>
            </div>
            {history.length > 0 && Object.entries(polls).length > 0 ?
                <div className="history_list custom-scroll">
                    {voteList && voteList.map((item,key) => (
                        <VoteHistoryItem key={key} poll={polls[item.pollId]} voteType={item.voteType}/>
                    ))}
                </div> :
                <h3 className="history_no-votes">No votes to show</h3>
            }
        </div>
    )
};

VoteHistoryList.propTypes = {
    polls: PropTypes.object,
    history: PropTypes.array
};
export default VoteHistoryList