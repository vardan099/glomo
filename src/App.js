import React, {useEffect, useState} from 'react';
import './App.scss';
import hookActions from './actions/hookActions';
import SportsPoll from './components/SportsPoll/SportsPoll';
import VoteHistoryList from "./components/VoteHistoryList/VoteHistoryList";

function App() {
    const [polls, setPolls] = useState({});
    const [randomPoll, setRandomPoll] = useState({});
    const [history, setHistory] = useState([]);

    useEffect(() => {
        const handleDataAfterFetch = (data) =>{
            const myData = {};
            data.forEach(pollEvent => {
                if (pollEvent.state !== "FINISHED") {
                    myData[pollEvent.objectId] = pollEvent
                }
            });
            setPolls(myData);
            updateRandomPollState(myData);
        };

        hookActions.getPolls(handleDataAfterFetch);
        const history = JSON.parse(localStorage.getItem('history'));
        if(history){
            setHistory(history)
        }
    }, []);



    const voteHandler = (pollId, voteType) => {
        const newState = [...history,{
            pollId,
            voteType,
            createdAt: Date.now()
        }];

        setHistory(newState);
        updateRandomPollState(polls);
        localStorage.setItem('history', JSON.stringify(newState));
    };

    const updateRandomPollState = (polls) => {
        if (polls) {
            const randomPoll = polls[[Object.keys(polls)[Math.floor(Math.random() * Object.keys(polls).length)]]];
            setRandomPoll(randomPoll)
        }
    };

    return (

        <div className="App w-100">
            <SportsPoll game={randomPoll} voteHandler={voteHandler}/>
            <hr className="h-line"/>
            <VoteHistoryList polls={polls} history={history}/>
        </div>
    );
}

export default App;
