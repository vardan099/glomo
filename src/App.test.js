import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import {shallow} from "enzyme/build";
import VoteHistoryList from "./components/VoteHistoryList/VoteHistoryList";
import SportsPoll from "./components/SportsPoll/SportsPoll";

it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
});

describe('Component Renders', () => {
    let wrapper;
    beforeEach(() => {
        wrapper = shallow(<App/>);
    });

    it('renders correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('Should render a VoteHistoryList Components', () => {
        const desc = wrapper.find(VoteHistoryList);
        expect(desc.length).toBe(1);
    });

    it('Should render a SportsPoll Components', () => {
        const desc = wrapper.find(SportsPoll);
        expect(desc.length).toBe(1);
    });

});

