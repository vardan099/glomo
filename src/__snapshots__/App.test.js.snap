// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Renders renders correctly 1`] = `ShallowWrapper {}`;
